addon/authenticators/oauth2-password-grant.js

```diff
@@ -336,6 +346,17 @@ export default BaseAuthenticator.extend({
   makeRequest(url, data, headers = {}) {
     headers['Content-Type'] = 'application/x-www-form-urlencoded';
 
+    if (this.get('sendClientIdAsQueryParam')) {
+      const clientId = this.get('clientId');
+      if (!isEmpty(clientId)) {
+        data['client_id'] = this.get('clientId');
+      }
+    }
+
+    if (data['grant_type'] === 'password' && this.get('refreshAccessTokens')) {
+      data['offline_token'] = true;
+    }
```

Comment:
This seems unrelated?

Reply:
I can open a second pull request if you like. It's actually required if you want access tokens to be refreshed by an Oauth2 server that's been implemented to spec. Generally you have to pass the offline_token as well as have the server allow for tokens to be refreshed.

Reply:
Hm, I can't find any reference to _"offline_token"_ in https://tools.ietf.org/html/rfc6749… Anyway, I'd put this in a separate PR for simplicity.

Reply:
https://tools.ietf.org/html/rfc6749#page-47

Reply:
I'll have to dig a bit more, but it looks it might not be a spec thing, but instead an opinionated option that many oauth2 services implement. The most common being offline_token in the body, but Auth0 implements it as a scope. 

I think it's important to look at adding because many require it.